src/node-uptimes/node-uptimes.service.ts

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { Injectable } from '@nestjs/common';
+import { User } from '@prisma/client';
+import { PrismaService } from '../prisma/prisma.service';
+import { NodeUptime } from '.prisma/client';
+
+@Injectable()
+export class NodeUptimesService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async upsert(user: User): Promise<NodeUptime | null> {
+    const now = new Date();
+    const oneHourAgo = new Date();
+    oneHourAgo.setHours(now.getHours() - 1);
+
+    return this.prisma.$transaction(async (prisma) => {
+      const nodeUptime = await this.get(user);
```

Comment:
Should we pass `prisma` into this?

Reply:
Do you normally not use dep injection to get services like this?

Reply:
For clarity, I wasn't using the transaction's `prisma` to do the `get`, I was using the one on the service which was likely not handling the transaction properly